datafusion/functions/Cargo.toml

```diff
@@ -57,8 +57,6 @@ path = "src/lib.rs"
 
 [dependencies]
 arrow = { workspace = true }
-arrow-array = { workspace = true }
```

Comment:
The point of the PR is to remove these two lines. The rest of the changes were driven by the compiler